### PR TITLE
Adding Lesson 16

### PR DESCRIPTION
**Pull request by lcoome**
Lecture and images

### CODE REVIEW

File: lec16-data-viz.Rmd [hunk 1]
```diff
@@ -1,4 +1,175 @@
...
+
+> 
+> ### Lesson outline
+>
+> - How to visualize data (review) (5 min)
+> - Saving your graphs (5 min)
+> - Review of plot choice (5 min)
+> - 2D histograms, square and hexagonal bins (10 min)
+> - Interactive and quick graphing (10 min)
+> - Graphing and accessability (5 min)
```
Comment (QuLogic) [inline lec16-data-viz.Rmd:23]: accessibility

File: lec16-data-viz.Rmd [hunk 2]
```diff
@@ -1,4 +1,175 @@
...
+> - Load `ggplot2` (`library("ggplot2")`).
+
+-----
+
+## Tufte's guidelines
+
+* Reduce non-data ink
+* Enhance the data ink
+
+Reduces the proportion of graphic’s ink devoted to the non-redundant display of data-information,
```
Comment (QuLogic) [inline lec16-data-viz.Rmd:36]: Dangling comma.

File: lec16-data-viz.Rmd [hunk 3]
```diff
@@ -1,4 +1,175 @@
...
+
+Avoid "chartjunk" - extraneous visual elements that detract from message.
+
+## Colour rules
+
+* Large background colours should be quiet, muted to let brighter colors stand out
+* To highlight some element of a figure, using a bright colour can be effective
+* However, if brightly colouring this aspect of your figure serves no purpose, leave it greyscale/plain
+* Here is a reference sheet with all the [colours and colour names in R](http://www.stat.columbia.edu/~tzheng/files/Rcolor.pdf)
```
Comment (QuLogic) [inline lec16-data-viz.Rmd:45]: https://stat.columbia.edu/~tzheng/files/Rcolor.pdf

**Review by QuLogic — COMMENTED** [review]
Mostly typos.

File: lec16-data-viz.Rmd [hunk 4]
```diff
@@ -1,4 +1,175 @@
...
+Wikipedia's definition of a heat map:
+
+["A heat map (or heatmap) is a graphical representation of data where the individual values contained in a matrix are represented as colors."](https://en.wikipedia.org/wiki/Heat_map)
+
+We've now added additional information to our graph and solved the saturation problem we encountered in our first graph. If you want to change the bin size (i.e. the size of the hexagons), you can do so as such:
+
+```{r}
+g <- ggplot(diamonds, aes(x = carat, y = price))
+g + 
+  geom_hex(bins = 10)
```
Comment (joelostblom) [inline lec16-data-viz.Rmd]: Could you change to `bins = 90`? The reason is that you get something similar to the scatter plot above for individual observations, combined with an indication of density where the plot is oversaturated:

image

File: lec16-data-viz.Rmd [hunk 5]
```diff
@@ -1,4 +1,175 @@
...
+
+[Documentation for square/rectangular heat maps of 2D bin counts](http://ggplot2.tidyverse.org/reference/geom_bin2d.html)
+
+## Making your graphs accessible
+
+Colour blindness is common in the population, and red-green colour blindness in particular affects 8% of men and 0.5% of women.
+
+Here are some guidelines for making your visualizations more accessible to people affected by colour blindness:
+
+Don't use jet heatmaps. Jet colourmaps are often the default heatmap used in many visualization packages (you've probably seen them before).
```
Comment (joelostblom) [inline lec16-data-viz.Rmd]: Don't use jet heatmaps `->` Don't use the jet rainbow colored theme for heatmaps

File: lec16-data-viz.Rmd [hunk 6]
```diff
@@ -1,4 +1,175 @@
...
+  geom_bin2d()
+```
+
+[Documentation for square/rectangular heat maps of 2D bin counts](http://ggplot2.tidyverse.org/reference/geom_bin2d.html)
+
+## Making your graphs accessible
+
+Colour blindness is common in the population, and red-green colour blindness in particular affects 8% of men and 0.5% of women.
+
+Here are some guidelines for making your visualizations more accessible to people affected by colour blindness:
```
Comment (joelostblom) [inline lec16-data-viz.Rmd]: Join this into the same paragraph as the above sentence and add a sentence:

> Colour blindness is common in the population, and red-green colour blindness in particular affects 8% of men and 0.5% of women. Guidelines for making your visualizations more accessible to people affected by colour blindness,  will in many cases also improve the interpretability of your graphs for people who have standard color vision. Here are a couple of examples:

File: lec16-data-viz.Rmd [hunk 7]
```diff
@@ -1,4 +1,175 @@
...
+
+Don't use jet heatmaps. Jet colourmaps are often the default heatmap used in many visualization packages (you've probably seen them before). 
+
+![A Jet Heatmap](./image/heatmap.png)
+
+Colour blind viewers are going to have a difficult time distinguishing the meaning of this heat map if some of the colours blend together.
+
+![Colour Blind](./image/colourblind.png)
+
+[PS - jet heatmaps aren't so great for other reasons, too!](https://jakevdp.github.io/blog/2014/10/16/how-bad-is-your-colormap/)
```
Comment (joelostblom) [inline lec16-data-viz.Rmd]: I would change to add some more info:

> The jet colormap should be avoided for other reasons, including that the sharp transitions between colors gives introduces visual threshold levels that do not represent the underlying continuous data, more details about jet can be found in [this blog post](https://jakevdp.github.io/blog/2014/10/16/how-bad-is-your-colormap/) and [this series of posts](https://mycarta.wordpress.com/2012/05/12/the-rainbow-is-dead-long-live-the-rainbow-part-1/). In general, when presenting continuous data, a perceptually uniform colormap is often the most suitable choice. This type of colormap ensures that equal steps in data are perceived as equal steps in color space. The human brain perceives changes in lightness as changes in the data much better than, for example, changes in hue. Therefore, colormaps which have monotonically increasing lightness through the colormap will be better interpreted by the viewer. More details and examples of such colormaps are available in the [matplotlib documentation](http://matplotlib.org/users/colormaps.html), and many of the core design principles are outlined in [this entertaining talk](https://www.youtube.com/watch?v=xAoljeRJ3lU). Most of these colormaps are [available in R](http://r4ds.had.co.nz/graphics-for-communication.html#fig:brewer). 

I know you don't have that much time, so the links are more included for the students as a reference.

**Review by joelostblom — CHANGES_REQUESTED** [review]
Looking good! Just a few comments.

File: lec16-data-viz.Rmd (hunk 4 above)
Comment (lcoome) [reply lec16-data-viz.Rmd]: yeah, good idea

File: lec16-data-viz.Rmd (hunk 6 above)
Comment (lcoome) [reply lec16-data-viz.Rmd]: changed

File: lec16-data-viz.Rmd (hunk 5 above)
Comment (lcoome) [reply lec16-data-viz.Rmd]: Usually that's what people mean when they say jet heatmap but I can add that in

File: lec16-data-viz.Rmd (hunk 7 above)
Comment (lcoome) [reply lec16-data-viz.Rmd]: Joel, I really think the wording of this is too dense for your average undergrad. I can add it and explain the issue of differences in brightness/luminosity/turning a jet heatmap greyscale though, if you like.

Comment (joelostblom) [reply lec16-data-viz.Rmd]: But they are not your average undergrads! They are 4th year research students who have already aced quizzes about more complex concepts than this earlier in our class.

Feel free to simplify when you speak about it during the lecture as long as the same message comes across, but I think it is good to keep the full content (potentially clarified) in the lecture as a resource.

File: lec16-data-viz.Rmd (hunk 5 above)
Comment (joelostblom) [reply lec16-data-viz.Rmd]: I should have said `jet rainbow colormap` instead of `jet rainbow colored theme`

File: lec16-data-viz.Rmd [hunk 8]
```diff
@@ -170,6 +168,5 @@ For more resources, [here](http://blog.usabilla.com/how-to-design-for-color-blin
 * [ggplot2 cheat sheet](https://www.rstudio.com/wp-content/uploads/2015/03/ggplot2-cheatsheet.pdf)
 * [r graph gallery for inspiration (not just limited to ggplot2 graphs)](https://www.r-graph-gallery.com/all-graphs/)
 
-*Parts of this lesson material were taken and modified from [Software
-Carpentry](https://software-carpentry.org) under their CC-BY copyright license. See
-their [lesson page](http://swcarpentry.github.io/r-novice-gapminder/08-plot-ggplot2/) for the original source.*
+
+
```
Comment (joelostblom) [inline lec16-data-viz.Rmd]: Remove newlines

File: lec16-data-viz.Rmd [hunk 9]
```diff
@@ -131,19 +131,17 @@ g +
...
...transitions between colors gives introduces visual threshold levels that do not represent the underlying continuous data, more details about jet can be found in [this blog post](https://jakevdp.github.io/blog/2014/10/16/how-bad-is-your-colormap/) and [this series of posts](https://mycarta.wordpress.com/2012/05/12/the-rainbow-is-dead-long-live-the-rainbow-part-1/). In general, when presenting continuous data, a perceptually uniform colormap is often the most suitable choice. This type of colormap ensures that equal steps in data are perceived as equal steps in color space. The human brain perceives changes in lightness as changes in the data much better than, for example, changes in hue. Therefore, colormaps which have monotonically increasing lightness through the colormap will be better interpreted by the viewer. More details and examples of such colormaps are available in the [matplotlib documentation](http://matplotlib.org/users/colormaps.html), and many of the core design principles are outlined in [this entertaining talk](https://www.youtube.com/watch?v=xAoljeRJ3lU). Most of these colormaps are [available in R](http://r4ds.had.co.nz/graphics-for-communication.html#fig:brewer).
```
Comment (joelostblom) [inline lec16-data-viz.Rmd]: First sentence: "...transitions between colors ~~gives~~ introduces visual threshold levels..."

**Review by joelostblom — APPROVED** [review]
Looks good apart from a typo in my previous suggestion.

**Pull request merged**